Makefile

```diff
@@ -173,6 +176,11 @@ $(LINK_CHECKER): $(TOOLS_DIR)/go.mod
 $(GO_APIDIFF): $(TOOLS_DIR)/go.mod
 	cd $(TOOLS_DIR) && go build -tags=tools -o $(GO_APIDIFF_BIN) github.com/joelanford/go-apidiff
 
+$(ENVSUBST): $(TOOLS_DIR)/go.mod
+	cd $(TOOLS_DIR) && go build -tags=tools -o $(ENVSUBST_BIN) github.com/a8m/envsubst/cmd/envsubst
```

Comment:
If I'm not wrong we should add a new import

https://github.com/kubernetes-sigs/cluster-api/blob/5596ab2a8d3f7dc14ca63e7f425df7c8f76fab3d/hack/tools/tools.go#L23

Also, why are we using `github.com/a8m/envsubst` instead of `github.com/drone/envsubst`? my concern is that there will be subtle differences between how those library behaves...
cc @wfernandes

Reply:
drone doesn't offer a binary, I think they are good enough in compatibility

Reply:
@benmoss drone doesn't offer a binary as part of it's releases but there is a `cmd/envsubst` in drone that can be built. So I believe we can build the binary if we needed to.

There are some subtle differences between `drone/envsubst` and `a8m/envsubst`. The one difference I know of is that a8m handles the differences with operators such as `:=, =, :-` whereas drone treats them all the same. Also drone documents that it doesn't support `:+` notation as a8m does.

Reply:
Oversight on the package differences, I'll grab `drone/envsubst` and build the binary